### ISSUE TITLE
Exception when trying to copy "Log Output" in UI cause of HTTP/2 with WebSockets

### ISSUE THREAD

**Issue by crdnb** [kubeshop/testkube#2978]
**Describe the bug**

We have an issue regarding WebSockets on OpenShift. We can not stream the logs from Testkube executors to the UI in real time. We use HAProxy-based Ingress Controllers on our platform and have enabled HTTP/2.
Using WebSockets with HTTP/2 enabled requires WebSocket support over HTTP/2.

**To Reproduce**

A client may connect to the Ingress Controller and negotiate HTTP/1.1, and the Ingress Controller may then connect to the application, negotiate HTTP/2, and forward the request from the client HTTP/1.1 connection using the HTTP/2 connection to the application. This poses a problem if the client subsequently tries to upgrade its connection from HTTP/1.1 to the WebSocket protocol, because the Ingress Controller cannot forward WebSocket to HTTP/2 and cannot upgrade its HTTP/2 connection to WebSocket. Consequently, if you have an application that is intended to accept WebSocket connections, it must not allow negotiating the HTTP/2 protocol or else clients will fail to upgrade to the WebSocket protocol.

**Expected behavior**
Is it possible that we can configure our backend service to not allow negotiating the HTTP/2 protocol? Another solution would be if it's possible to terminate TLS on our backend pod. Is it possible to inject a certificate into the server pod and enable TLS on it?

**Screenshots**
If applicable, add CLI commands/output to help explain your problem.
 
**Additional context**
<https://docs.openshift.com/container-platform/4.11/networking/ingress-operator.html#nw-http2-haproxy_configuring-ingress>

**Comment by vsukhin** [comment]
@dejanzele any ideas?

**Comment by dejanzele** [comment]
Hi @crdnb,

Thanks for raising an issue about this.

We plan on adding support to have the API configurable if you want to run it on a non-secure port, or provide a certificate & key file (most convenient way is to mount it from a Kubernetes `kubernetes.io/tls` secret) and run a HTTPS server in the application. That way TLS can be terminated on app level, and we would achieve e2e HTTP2 connection (Ingresses usually downgrade connection to HTTP1.1 if connection is not on a secure port).

@TheBrunoLopes can you please coordinate when could we release this feature?

**Comment by TheBrunoLopes** [comment]
@dejanzele is this complex to implement ?

**Comment by TheBrunoLopes** [comment]
Hello @crdnb is this still a blocker for you ?
We are investigating this issue for the current release

**Comment by crdnb** [comment]
Hi @TheBrunoLopes 

Thank you for your message.

I just double checked it and websockets are working now in our environment, even with an HTTP2 ingress controller.
I did some research and found out, that we switched the route from HTTP to TLS with edge termination. To be honest, we do not understand at the moment, why this is working. We would expect to need TLS with passthrough termination to have it working, but it realldy works with edge termination.

Long story short: This is currently no longer a problem for us.

Thank you anyways!
